parser.py

```diff
@@ -483,52 +483,75 @@ def parse_direct_messages(data_folder, username, users, user_id_URL_template, dm
             dm_user_ids.add(user1_id)
             dm_user_ids.add(user2_id)
     lookup_users(list(dm_user_ids), users)
-    # Parse the DMs
-    num_written_messages = 0
-    long_conversations = []
+    # Parse the DMs and store the messages in a dict
+
+    conversations_messages = {}
```

Comment:
`conversations_messages = defaultdict(list)`

and then you don't need to check whether `other_user_id` is already present on L514